From c621d852016d113f3e1ae9509fc42f279d6fe19f Mon Sep 17 00:00:00 2001
From: Alexandre Torgue <alexandre.torgue@foss.st.com>
Date: Thu, 13 Oct 2022 17:04:00 +0200
Subject: [PATCH 0144/1141] arm64: dts: st: Enable ACM block on TSN switch on
 stm32mp257

Add ACM node and adapt the glue layer to enable the switch acceleration
module on stm32mp257.

Signed-off-by: Alexandre Torgue <alexandre.torgue@foss.st.com>
Change-Id: Id8ef708981f085de792cdebd3ef6781bcc34a342
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/271273
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Christophe ROULLIER <christophe.roullier@st.com>
Reviewed-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/304910
Tested-by: Christophe ROULLIER <christophe.roullier@st.com>
---
 arch/arm64/boot/dts/st/stm32mp257.dtsi | 29 ++++++++++++++++++++++----
 1 file changed, 25 insertions(+), 4 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp257.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp257.dtsi
@@ -12,12 +12,15 @@
 				#address-cells  = <1>;
 				#size-cells     = <1>;
 				compatible = "st,stm32-deip";
-				clock-names = "ethsw-bus-clk", "ethsw-clk";
+				clock-names = "ethsw-bus-clk", "ethsw-clk",
+					      "ethswacmcfg-bus-clk", "ethswacmmsg-bus-clk";
 				clocks = <&rcc CK_BUS_ETHSW>,
-					 <&rcc CK_KER_ETHSW>;
-
+					 <&rcc CK_KER_ETHSW>,
+					 <&rcc CK_BUS_ETHSWACMCFG>,
+					 <&rcc CK_BUS_ETHSWACMMSG>;
 				st,syscon = <&syscfg 0x3800>;
-				ranges = <0x4c000000 0x4c000000 0x2000000>;
+				ranges = <0x4c000000 0x4c000000 0x2000000>,
+					 <0x4b000000 0x4b000000 0xc0000>;
 				feature-domains = <&rifsc STM32MP25_RIFSC_ETHSW_DEIP_ID>;
 				power-domains = <&CLUSTER_PD>;
 				status = "disabled";
@@ -27,6 +30,24 @@
 					reg = <0x4c000000 0x2000000>;
 					interrupts = <GIC_SPI 250 IRQ_TYPE_LEVEL_HIGH>;
 				};
+
+				acm@4b000000 {
+					compatible = "ttt,acm-4.0";
+					reg = <0x4b000000 0x00400>,
+					      <0x4b010000 0x10000>,
+					      <0x4b030000 0x10000>,
+					      <0x4b050000 0x10000>,
+					      <0x4b060000 0x20000>,
+					      <0x4b080000 0x40000>;
+					reg-names = "CommonRegister",
+						    "Bypass1",
+						    "Bypass0",
+						    "Redundancy",
+						    "Scheduler",
+						    "Messagebuffer";
+					buffers = <32>;
+					ptp_worker = <&deip_sw0>;
+				};
 			};
 		};
 	};
